From 2689e70bc9b37f3ed2cdaca2e38d2b433f20fa7c Mon Sep 17 00:00:00 2001
From: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
Date: Fri, 20 Dec 2024 13:27:08 -0300
Subject: [PATCH] csi-powerscale: enabling csm-authorization

This patch consists of:
- change the format of the values in templates and changing the
  values.yaml for the user be able to enable csm-authorization.
- Change some authorization secrets to use the release name as prefix.
- Adds authorization-config template to create the secret used by
  the csm-observability/csm-authorization

Signed-off-by: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
---
 .../templates/authorization-config.yaml       | 20 +++++++++++++++++++
 charts/csi-isilon/templates/controller.yaml   |  8 ++++----
 charts/csi-isilon/templates/node.yaml         |  8 ++++----
 3 files changed, 28 insertions(+), 8 deletions(-)
 create mode 100644 charts/csi-isilon/templates/authorization-config.yaml

diff --git a/charts/csi-isilon/templates/authorization-config.yaml b/charts/csi-isilon/templates/authorization-config.yaml
new file mode 100644
index 0000000..ec6702d
--- /dev/null
+++ b/charts/csi-isilon/templates/authorization-config.yaml
@@ -0,0 +1,20 @@
+{{- if or .Values.authorization.enabled .Values.csm.authorization }}
+{{- if .Values.secret }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Release.Name }}-authorization-config
+  namespace: {{ .Release.Namespace }}
+stringData:
+  config: |
+    {{- range .Values.secret.arrays }}
+    - username: {{ .username }}
+      password: {{ .password }}
+      intendedEndpoint: {{ .endpoint }}
+      endpoint: {{ .authorizationEndpoint | default "https://localhost:9400" }}
+      systemID: {{ .clusterName }}
+      skipCertificateValidation: {{ .skipCertificateValidation | default true }}
+      isDefault: {{ .isDefault | default false }}
+    {{- end }}
+{{- end }}
+{{- end }}
\ No newline at end of file
diff --git a/charts/csi-isilon/templates/controller.yaml b/charts/csi-isilon/templates/controller.yaml
index 981c1bb..4dd50dd 100644
--- a/charts/csi-isilon/templates/controller.yaml
+++ b/charts/csi-isilon/templates/controller.yaml
@@ -136,7 +136,7 @@ metadata:
   name: {{ .Release.Name }}-controller
   namespace: {{ .Release.Namespace }}
   {{- if hasKey .Values "authorization" }}
-  {{- if eq .Values.authorization.enabled true }}
+  {{- if or .Values.authorization.enabled .Values.csm.authorization }}
   annotations:
     com.dell.karavi-authorization-proxy: "true"
   {{ end }}
@@ -473,7 +473,7 @@ spec:
             - name: csi-isilon-config-params
               mountPath: /csi-isilon-config-params
         {{- if hasKey .Values "authorization" }}
-        {{- if eq .Values.authorization.enabled true }}
+        {{- if or .Values.authorization.enabled .Values.csm.authorization }}
         - name: karavi-authorization-proxy
           imagePullPolicy: {{ .Values.imagePullPolicy }}
           image: {{ required "Must provide the authorization sidecar container image." .Values.authorization.image }}
@@ -523,10 +523,10 @@ spec:
           configMap:
              name: {{ .Release.Name }}-config-params
         {{- if hasKey .Values "authorization" }}
-        {{- if eq .Values.authorization.enabled true }}
+        {{- if or .Values.authorization.enabled .Values.csm.authorization }}
         - name: karavi-authorization-config
           secret:
-            secretName: karavi-authorization-config
+            secretName: {{ .Release.Name }}-authorization-config
         - name: proxy-server-root-certificate
           secret:
             secretName: {{ .Release.Name }}-proxy-server-root-certificate
diff --git a/charts/csi-isilon/templates/node.yaml b/charts/csi-isilon/templates/node.yaml
index 6c7d965..86f0f50 100644
--- a/charts/csi-isilon/templates/node.yaml
+++ b/charts/csi-isilon/templates/node.yaml
@@ -66,7 +66,7 @@ metadata:
   name: {{ .Release.Name }}-node
   namespace: {{ .Release.Namespace }}
   {{- if hasKey .Values "authorization" }}
-  {{- if eq .Values.authorization.enabled true }}
+  {{- if or .Values.authorization.enabled .Values.csm.authorization }}
   annotations:
     com.dell.karavi-authorization-proxy: "true"
   {{ end }}
@@ -256,7 +256,7 @@ spec:
             - name: driver-path
               mountPath: /csi
         {{- if hasKey .Values "authorization" }}
-        {{- if eq .Values.authorization.enabled true }}
+        {{- if or .Values.authorization.enabled .Values.csm.authorization }}
         - name: karavi-authorization-proxy
           imagePullPolicy: {{ .Values.imagePullPolicy }}
           image: {{ required "Must provide the authorization sidecar container image." .Values.authorization.image }}
@@ -324,10 +324,10 @@ spec:
           configMap:
              name: {{ .Release.Name }}-config-params
         {{- if hasKey .Values "authorization" }}
-        {{- if eq .Values.authorization.enabled true }}
+        {{- if or .Values.authorization.enabled .Values.csm.authorization }}
         - name: karavi-authorization-config
           secret:
-            secretName: karavi-authorization-config
+            secretName: {{ .Release.Name }}-authorization-config
         - name: proxy-server-root-certificate
           secret:
             secretName: {{ .Release.Name }}-proxy-server-root-certificate
-- 
2.34.1
